src/Data/Ring/Extra.purs

```diff
@@ -0,0 +1,12 @@
+module Data.Ring.Extra where
```

Comment:
From the outside, it makes little sense to put this function into `Data.Ring.Extra` rather than just putting it into `Data.Ring`. What if we move `abs` and `signum` from `Data.Ord` into `Data.Ring`? I think that would mean that `Data.Ord` no longer needs to import `Data.Ring`, which would allow us to use `>=` here.

I think we should basically inline the definitions of `power` and `Additive` too, since importing either of those in `Data.Ring` will also cause a cycle.

Reply:
Sounds like a breaking change.

Reply:
That'd be a breaking change though, if `Data.Ord` doesn't export `abs`/`signum` anymore.

Reply:
Correct, so this would have to wait until the next round of breaking changes. Even though it is breaking, I prefer it over adding a new `Data.Ring.Extra` module because:
* the API surface makes more sense,
* we'd end up wanting to merge Data.Ring.Extra back into Data.Ring at some point in the future, which would also be breaking, and
* I think it makes more sense to put `ord` and `signum` into Data.Ring anyway

Reply:
Ord also depends on `negate` in its usage of `-1`:
```purescript
instance ordArray :: Ord a => Ord (Array a) where
  compare = \xs ys -> compare 0 (ordArrayImpl toDelta xs ys)
    where
    toDelta x y =
      case compare x y of
        EQ -> 0
        LT -> 1
        GT -> -1 -- here it is
```